### PR TITLE
[cmake] Adding c++14 support

### PR DESCRIPTION
**Pull request by greshilov**
Note, that ```my::make_unique``` from ```base/stl_add.hpp``` will be changed in next PR's.
I also suggest edit our ```docs```, especially ```CPP_STYLE.md```, ```INSTALL.md```, ```CONTRIBUTING.md```. 
If there are any suggestions about c++14 features usage it makes sense to describe them in ```CPP_STYLE.md```.

### CODE REVIEW

**Comment by greshilov** [comment]
JTTP

File: skin_generator/CMakeLists.txt [hunk 1]
```diff
@@ -1,7 +1,6 @@
 project(skin_generator_tool)
 
 add_compile_options(
-  "-std=c++11"
   "-Wno-c++11-narrowing"
```
Comment (VladiMihaylenko) [inline skin_generator/CMakeLists.txt]: why not c++14?

**Comment by bykoianko** [comment]
Мы должны явно указать, что хотим собираться C++14?

**Comment by bykoianko** [comment]
Что-то вроде "-std=c++14"?

File: skin_generator/CMakeLists.txt (hunk 1 above)
Comment (greshilov) [reply skin_generator/CMakeLists.txt]: ```set(CMAKE_CXX_STANDARD 14)``` adds this compile option by default:
https://cmake.org/cmake/help/v3.1/variable/CMAKE_CXX_STANDARD.html

Comment (VladiMihaylenko) [reply skin_generator/CMakeLists.txt]: got it, thanks

Comment (bykoianko) [reply skin_generator/CMakeLists.txt]: А что означает "-Wno-c++11-narrowing"?

File: std/unique_ptr.hpp [hunk 2]
```diff
@@ -6,14 +6,7 @@
...
 using std::unique_ptr;
-
-/// @todo(y): replace this hand-written helper function by
-/// std::make_unique when it will be available in C++14
-template <typename T, typename... Args>
-unique_ptr<T> make_unique(Args &&... args)
-{
-  return unique_ptr<T>(new T(std::forward<Args>(args)...));
-}
+using std::make_unique;
```
Comment (ygorshenin) [inline std/unique_ptr.hpp]: Finally, thanks!

**Comment by bykoianko** [comment]
Let's replace my::make_unique for base/stl_add.hh

File: map/mwm_tests/CMakeLists.txt [hunk 3]
```diff
@@ -1,10 +1,5 @@
 project(mwm_tests)
 
-add_compile_options(
-  "-Wall"
```
Comment (syershov) [inline map/mwm_tests/CMakeLists.txt]: Don't remove this option please

Comment (greshilov) [reply map/mwm_tests/CMakeLists.txt]: This compile option is added by default in root ```omim/CMakeLists.txt:240```
```cmake
# Only options related to warnings should be placed here.
# Other options should be set before all add_subdirectory calls.
add_compile_options(
  "-Wall"
)
```

File: skin_generator/CMakeLists.txt (hunk 1 above)
Comment (greshilov) [reply skin_generator/CMakeLists.txt]: Я попробовал собирать с/без этой опции на gcc/clang - она не дает никакого эффекта.

**Comment by greshilov** [comment]
@bykoianko we'll defintely replace ```my::make_unique``` in next PR, I was advised to divide this changes in two parts :)

**Comment by greshilov** [comment]
JTTP

**Comment by greshilov** [comment]
JTTP finally

**Comment by greshilov** [comment]
JTTP

File: map/mwm_tests/CMakeLists.txt (hunk 3 above)
Comment (syershov) [reply map/mwm_tests/CMakeLists.txt]: ok

**Comment by mgsergio** [comment]
LGTM!

**Pull request merged**